libtcmu.c

```diff
@@ -383,18 +383,192 @@ static void tcmu_cdb_debug_info(const struct tcmulib_cmd *cmd)
                 free(buf);
 }
 
-static void *tcmu_cmdproc_thread(void *arg)
+static void _cleanup_mutex_lock(void *arg)
+{
+        pthread_mutex_t *l = arg;
+        pthread_mutex_unlock(l);
```

Comment:
It's just void * so just pass it directly here and below.

pthread_mutex_unlock(arg);

Also no need to cast to a void * in the pthread_cleanup_push calls below.